### PR TITLE
Add an ability to specify CORS headers

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

**Comment by mod** [comment]
I prefer using https://github.com/ruby-grape/grape#cors
Rack::Cors middleware as it's natively supported by grape
I should be smart behavior with OPTIONS requests.

File: Gemfile [hunk 1]
```diff
@@ -53,6 +53,8 @@ gem 'cash-addr', '~> 0.1.1', require: 'cash_addr'
 gem 'digest-sha3', '~> 1.1.0'
 gem 'scout_apm', '~> 2.4', require: false
 gem 'peatio', '~> 0.4.4'
+gem 'rack-cors', '~> 1.0.2', :require => 'rack/cors'
```
Comment (ysv) [inline Gemfile]: require: false

Because we use it only in config.ru

File: lib/cors/validations.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Provides CORS variables validation.
+module CORS::Validations
+  Error = Class.new(StandardError)
+
+  class << self
+    def validate_origins(origins)
+      if origins.include?("*")
```
Comment (ysv) [inline lib/cors/validations.rb]: May be origins=='*'

File: lib/cors/validations.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        Rails.logger.info{ "WARNING: Usinng API_CORS_ORIGIN as '*'" }
```
Comment (ysv) [inline lib/cors/validations.rb]: WARNING: API_CORS_ORIGIN is set to '*'

File: lib/cors/validations.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          raise CORS::Validations::Error, "Set right origin domain value instead of #{origin}"
+        end
+      end
+    end
+
+    def validate_max_age(max_age)
+      if max_age.present? && max_age.match?(/^[0-9]*$/) 
+        max_age
+      else
+        Rails.logger.info{ "WARNING: Incorect or missing API_CORS_MAX_AGE value. Set default value: 3600" }
```
Comment (ysv) [inline lib/cors/validations.rb]: WARNING: Incorect or missing API_CORS_MAX_AGE value. Using default value: 3600

**Review by ysv — CHANGES_REQUESTED** [review]

File: lib/cors/validations.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# Provides CORS variables validation.
+module CORS::Validations
+  Error = Class.new(StandardError)
+
+  class << self
+    def validate_origins(origins)
+      origins.split(',').each_with_object([]) do |origin, domains|
+        if origin == '*'
+          Rails.logger.info{ "WARNING: API_CORS_ORIGIN is set to '*'" }
+          return '*'
```
Comment (ysv) [inline lib/cors/validations.rb]: As for me the most logical way is
```ruby 
if origin == '*'
  Rails.logger.info{ "WARNING: API_CORS_ORIGIN is set to '*'" }
￼   return '*'
origins.split(',').each_with_object([]) do ...
```

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Pull request merged**